### DISCUSSION TITLE
Regex Value Mapping for numbers

### DISCUSSION THREAD

**Discussion by modularTaco** [grafana/grafana#42819] [Dashboards, Panels & Visualizations]
Hi,

i tried to use the regex value mapping with a number field and was frustrated that nothing worked. 
Until i took a look at the code ([valueMappings.ts#L55](https://github.com/grafana/grafana/blob/beb3a8f02d2681e867cfab845661d81c7ea3ebaa/packages/grafana-data/src/utils/valueMappings.ts#L55)) and saw that only values of type string are supported.

So i think Grafana should either:
- Show a warning if the regex value mapping is used for an unsupported type of field (different from string)
- Work with different types (e.g. numbers) as well (could be converted to string prior regex matching, but using the original number value for calculations, graphing (in the stat panel) etc)

My use case is the following:  
I want that a value in a stat panel is always signed. So positive values should be signed with a `+` symbol. (I tried the regex: `^[0-9]+$` and the substitution `+ $1`).  
Another use case would be prefixing positive values with an up arrow (`🠕`) and negative values with a down arrow (`🠗`)

**Comment by ramoslomas** [comment]
Hi, 

did you find a way to do it?

 i have a similar use case with a gauge and if i do a transformation to convert my data to string so i can match the regex the gauge just says no data. Thank you

**Comment by precisionconage** [comment]
I also want this functionality, or alternatively, a way to show the entire value rather than abbreviating to e.g. $1.05K for currency. My plan was something like this:
1. Use locale formatting to format the number with commas, spaces, dots etc. according to locale
2. Use regex `(.*)` with replacement `$$$1` to add a dollar sign at the start

Agreed that this limitation should at least be documented though, because right now the only way to figure out why it's failing is to look at the source code.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
